### DISCUSSION TITLE
Overlay segments and normal segments exist at the same time.

### DISCUSSION THREAD

**Discussion by jason30704** [bbc/peaks.js#544] [Q&A]
Hello,
If I want to use overlay segments and normal segments at the same time in waveform, is it possible?

**Comment by chrisn** [comment]
Currently no. We would need to add more per-segment options to make this work. Something like:

```js
peaks.segments.add({
  startTime: 1.0,
  endTime: 2.0,
  overlay: false, // overrides default options.segmentOptions.overlay
  editable: true
});
```

**Reply by jason30704** [reply]
I tried using the following code, but it doesn't seem to override successfully.
![image](https://github.com/user-attachments/assets/fe162c34-fd66-4362-aa58-a70f672663ae)
![image](https://github.com/user-attachments/assets/b9d03e14-32eb-4efb-99c8-ee8ece59ff8f)

The result will look like this. The waveform color of the segment cannot be changed successfully either.
![image](https://github.com/user-attachments/assets/2d9e9737-ab50-4f20-9b65-d60c53101b31)

**Reply by chrisn** [reply]
The `overlay: false` option isn't implemented yet. My previous comment was just an idea of how to solve this.

**Reply by jason30704** [reply]
Got it, thank you.
Is it also expected that the waveformColor of the segment cannot be changed?

**Reply by chrisn** [reply]
You can use  `segment.update({ color: ... })` to update the segment color.

If you have overlay segments, this changes the overlay color and you cannot change the waveform color.

If you have non-overlay segments, this changes the waveform color.

**Reply by jason30704** [reply]
Thanks, use segment.update() is worked.
It seems that setting waveformColor in segmentOptions has no effect. 
![image](https://github.com/user-attachments/assets/075cb604-073b-4f28-8394-93c1b3da7ce5)

**Reply by chrisn** [reply]
It should set the waveform color, but it only applies with `overlay: false`.
